pkg/objstore/azure/helpers.go

```diff
@@ -38,20 +38,31 @@ func getAzureStorageCredentials(conf Config) (blob.Credential, error) {
 	if conf.MSIResource != "" {
 		msiConfig := auth.NewMSIConfig()
 		msiConfig.Resource = conf.MSIResource
-
 		azureServicePrincipalToken, err := msiConfig.ServicePrincipalToken()
 		if err != nil {
 			return nil, err
 		}
 
-		// Get a new token.
+		var tokenRefresher blob.TokenRefresher = func(credential blob.TokenCredential) time.Duration {
+			err := azureServicePrincipalToken.Refresh()
+			if err != nil {
+				// Retry later as the error can be related to API throttling
+				return 30 * time.Second
```

Comment:
Its annoying that there is no good error handling in this library. It might be good to print the error to make it visible. Exiting on an error is not a good solution as the error good be temporary of related to an API rate limit, so its a lot better to retry. Any thoughts?

Reply:
Yes, We should report that in the library GH issues ASAP.

Maybe there are different error codes, but not documented? 🤗 

Do you think this will work for now? Happy to merge, for now.

Reply:
I added a log to make it visible, hopefully it is enough for now.

Reply:
yup